src/routes.js

```diff
@@ -0,0 +1,62 @@
+'use strict'
+
+const fs = require('fs')
+const path = require('path')
+
+const logger = require('./logger')
+const {expectedEndpointsDirectories} = require('./constants')
+
+exports.createRoutes = router => {
+  validateDirectoryStructure()
+  setUpRoutes(router)
+}
+
+const validateDirectoryStructure = () => {
+  if (!fs.existsSync(path.resolve(__dirname, '..', 'endpoints'))) {
+    logger.error('Directory "endpoints" not found in project root')
+    process.exit(1)
+  }
+
+  const routeDirectories = fs.readdirSync(
+    path.resolve(__dirname, '..', 'endpoints')
+  )
+  let correctDirectoryStructure = true
+  routeDirectories.forEach(directory => {
+    expectedEndpointsDirectories.forEach(expectedFile => {
+      if (
+        !fs.existsSync(
+          path.resolve(__dirname, '..', 'endpoints', directory, expectedFile)
+        )
+      ) {
+        logger.error(
+          `Missing file "${expectedFile}" in directory "${directory}"`
+        )
+        correctDirectoryStructure = false
+      }
+    })
+  })
+  if (!correctDirectoryStructure) {
+    logger.error('Add required files')
+    process.exit(1)
+  }
+}
+
+const setUpRoutes = router => {
+  const routeDirectories = fs.readdirSync(
+    path.resolve(__dirname, '..', 'endpoints')
+  )
+
+  routeDirectories.forEach(directory => {
+    const metaFile = fs.readFileSync(
+      path.resolve(__dirname, '..', 'endpoints', directory, 'meta.json')
+    )
+    const metaJson = JSON.parse(metaFile)
+    router.post(metaJson.endpoint.pattern, ctx => {
+      ctx.body = `${directory} path exists!`
+    })
+    logger.info(`New Route added: ${directory} at path ${metaJson.endpoint.pattern}`)
+  })
+}
+
+exports.setUpRoutes
+exports.validateDirectoryStructure
```

Comment:
Why are these functions being exported (will they be called from outside the module) ?

Reply:
For the none existant test framework... Maybe I'll add soem tests

Reply:
We should be able to test these functions by calling `createRoutes` and checking against the injected `router` object that the routes have been created.
The functions shouldn't be called from outside the module, so shouldn't be exported.